src/Chapter4.hs

```diff
@@ -673,6 +706,23 @@ Specifically,
  ❃ Implement the function to convert Tree to list
 -}
 
+data Tree a = Leaf a | Node (Tree a) (Tree a)
+
+instance Functor Tree where
+    fmap :: (a -> b) -> Tree a -> Tree b
+    fmap f (Leaf a) = Leaf (f a)
+    fmap f (Node tl tr) = Node (fmap f tl) (fmap f tr)
+
+reverseTree :: Tree a -> Tree a
+reverseTree (Node (Leaf a) (Leaf b)) = Node (Leaf b) (Leaf a)
+reverseTree (Node (Leaf a) tr) = Node (reverseTree tr) (Leaf a)
+reverseTree (Node tl (Leaf a)) = Node (Leaf a) (reverseTree tl)
+reverseTree (Node tl tr) = Node (reverseTree tr) (reverseTree tl) 
```

Comment:
It looks like pattern-matching is not exhaustive in this function. You need to cover the case when you have only the `Leaf` constructor. Once you cover such case, you will be able to see, how you can simplify your function by handling only two cases :slightly_smiling_face: